package GuindexStatsIf;

// Message to send to StatsServer when new price is added
message NewGuinnessStatsRequest {
    required int32 guinnessId = 1;
}

// Message to send to StatsServer when new pub is added
message NewPubStatsRequest {
    required int32 pubId  = 1;
}

// Message to send to StatsServer when pub is marked as closed
message PubClosedStatsRequest {
    required int32 pubId = 1;
}

// Message to send to StatsServer when pub is marked as not serving Guinness
message PubNotServingGuinnessStatsRequest {
    required int32 pubId = 1;
}

message GuindexStatsIfMessage {
    optional NewGuinnessStatsRequest newGuinnessStatsRequest = 1; 
    optional NewPubStatsRequest newPubStatsRequest = 2;
    optional PubClosedStatsRequest pubClosedStatsRequest = 3;
    optional PubNotServingGuinnessStatsRequest pubNotServingGuinnessStatsRequest = 4;
}
